rs/index/src/optimizers/engine.rs

```diff
@@ -27,10 +28,19 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> OptimizerEngine<Q> {
         if segments.len() < 2 && optimizing_type == OptimizingType::Merge {
             return Ok("".to_string());
         }
+        if segments.len() > 1 && optimizing_type == OptimizingType::Vacuum {
+            return Ok("".to_string());
+        }
```

Comment:
I think let's move these checks to their corresponding optimizer. so inside `VacuumOptimizer`, you'll have this check, and if so, return an  `Err`

Reply:
do the same for the Merge one above